### PR TITLE
CIViC

### PR DESCRIPTION
**Pull request by varvarenius**
CIViC load stage for checks and merge

### CODE REVIEW

File: CIViC/load_stage.sql [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+       s.vocabulary_id,
+       s.concept_class_id,
+       s.standard_concept,
+       s.concept_code,
+       s.valid_start_date,
+       s.valid_end_date ,
+      CASE WHEN s.concept_code is null then 'D' else null end as invalid_reason
+from s
+FULL OUTER JOIN concept c ON c.concept_code = s.concept_code -- already existing LOINC concepts
+    	and c.vocabulary_id = 'CIViC'
```
Comment (Alexdavv) [inline CIViC/load_stage.sql:90]: Why do we look back to the existing CIVIC concepts?

File: CIViC/load_stage.sql [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+* Authors: Varvara Savitskaya, Vlad Korsik
+* Date: 2022
+**************************************************************************/
+
+DO $_$
+BEGIN
+	PERFORM VOCABULARY_PACK.SetLatestUpdate(
+	pVocabularyName			=> 'CIViC',
+	pVocabularyDate			=> current_date,
+	pVocabularyVersion		=> 'CIViC'||TO_CHAR(CURRENT_DATE,'YYYYMMDD'),
```
Comment (Alexdavv) [inline CIViC/load_stage.sql:25]: Why is it a current date? Don't we have a date/version coming from the source?

File: CIViC/load_stage.sql [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+where variant ~'([A-Z][1-9]*[A-Z])'
+and variant!~*'expression|amplification|truncation|truncating|wildtype|mutation|loss|methylation|polymorphism|HOMOZYGOSITY|translocation|PHOSPHORYLATION|deletion|function|shift|alteration|tandem|serum|alternative|REARRANGEMENT|MISLOCALIZATION|and|INACTIVATION|DOMAIN'
+and variant~*'rs'
+and variant!~'::');
+
+
+--insert into concept_stage
+insert into concept_stage
+with s as (SELECT DISTINCT
+       COALESCE(c.concept_id, NULL::int)  as concept_id,
```
Comment (Alexdavv) [inline CIViC/load_stage.sql:63]: We never populate the concept_id field in the stage tables. Generic_update function does it

**Review by Alexdavv — CHANGES_REQUESTED** [review]
In general, a temporary civic_source table is not a working solution looking forward:
- it's not a source, it's something processed;
- it doesn't help in any way, because the INSERTS into the stage tables don't add any substantial logic, but become too weird and too deep embedded.
@hardhouse What do you think?

File: CIViC/load_stage.sql [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+from sources.civic_variantsummaries
+where variant ~'([A-Z][1-9]*[A-Z])'
+and variant!~*'expression|amplification|truncation|truncating|wildtype|mutation|loss|methylation|polymorphism|HOMOZYGOSITY|translocation|PHOSPHORYLATION|deletion|function|shift|alteration|tandem|serum|alternative|REARRANGEMENT|MISLOCALIZATION|and|INACTIVATION|DOMAIN'
+and variant~*'rs'
+and variant!~'::');
+
+
+--insert into concept_stage
+insert into concept_stage
+with s as (SELECT DISTINCT
```
Comment (hardhouse) [inline CIViC/load_stage.sql:62]: distinct is not needed here

File: CIViC/load_stage.sql [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+       COALESCE(c.concept_id, NULL::int)  as concept_id,
+       trim(substr(n.concept_name,1,255)) AS concept_name,
```
Comment (hardhouse) [inline CIViC/load_stage.sql:64]: don't do that, use vocabulary_pack.CutConceptName() (see our "SQL technical guide, common problems and tricks", p.1)

File: CIViC/load_stage.sql [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+       'Measurement' AS domain_id,
+       n.vocabulary_id AS vocabulary_id,
+       'Variant' AS concept_class_id,
+       NULL AS standard_concept,
+       n.concept_code AS concept_code,
+      COALESCE(c.valid_start_date, v.latest_update)   as valid_start_date, -- defines valid_start_date  based on previous vocabulary run
```
Comment (hardhouse) [inline CIViC/load_stage.sql:70]: if valid_start_date=latest_update, the generic [will leave the old date](https://github.com/OHDSI/Vocabulary-v5.0/blob/master/working/generic_update.sql#L121)

File: CIViC/load_stage.sql [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+       COALESCE(c.valid_end_date,TO_DATE('20991231', 'yyyymmdd'))  as valid_end_date -- defines valid_end_date  based on previous vocabulary run
```
Comment (hardhouse) [inline CIViC/load_stage.sql:71]: if the concept is alive in the source, but D or U in the concept, then why should it remain deprecated?
why do we need the concept table here at all?

File: CIViC/load_stage.sql [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+FROM civic_source n
+JOIN vocabulary v ON v.vocabulary_id = 'CIViC'
+LEFT JOIN concept c ON c.concept_code = n.concept_code -- already existing LOINC concepts
+    	AND c.vocabulary_id = 'CIViC')
+
+SELECT distinct
```
Comment (hardhouse) [inline CIViC/load_stage.sql:77]: looks like a useless query

File: CIViC/load_stage.sql [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+where s.concept_code is not null
+;
+
+
+-- insert synonyms
+insert into concept_synonym_stage
+select *
+from (
+select
+DISTINCT NULL::INT as synonym_concept_id,
```
Comment (hardhouse) [inline CIViC/load_stage.sql:100]: is a distinct necessary here?

File: CIViC/load_stage.sql [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+hgvs AS synonym_name,
+cs.concept_code as synonym_concept_code,
+cs.vocabulary_id AS synonym_vocabulary_id,
+4180186 as language_concept_id
+from civic_source a
+join concept_stage cs on cs.concept_code = a.concept_code
+) r
+where synonym_name is not null
```
Comment (hardhouse) [inline CIViC/load_stage.sql:108]: how can this field be null?

**Review by hardhouse — CHANGES_REQUESTED** [review]

**Comment by hardhouse** [comment]
1. please, resolve conflicts (CIViC/create_source_tables.sql already exists in master). so just remove yours from this PR
2. add a number to each item with comments
3. add all missing truncate of stage tables
4. why do you write "cast(variant_id as varchar(5))", because i asked in advance, whether this field should be made int or not?
5. why do you use distinct+union? please read our "SQL technical guide, common problems and tricks", p.6
6. do not abuse regular expressions, for example, instead of "variant!~*'rs'" you need to write "variant not ilike '%rs%'", it works faster. same for "::"
7. add clean-up (drop civic_source)
8. add the last line with standard text "-- At the end, the three tables concept_stage..."

**Comment by varvarenius** [comment]
@hardhouse why do you write "cast(variant_id as varchar(5))", because i asked in advance, whether this field should be made int or not? - 
Type of source data is int, but we use this field as concept_code, which should be varchar. Is it worth to pull the source again with another type of this field or not?

**Review by hardhouse — APPROVED** [review]

**Review by Alexdavv — APPROVED** [review]

**Pull request merged**